### DISCUSSION TITLE
Add entities to hold the preset temperature and change configuration menu

### DISCUSSION THREAD

**Discussion by jmcollin78** [jmcollin78/versatile_thermostat#404] [Announcements]
# DO NOT INSTALL THIS RELEASE IF YOU ARE NOT INVITED TO DO SO

The release aims to be tested in a real environment.

This major release implements:
1. Add some entities to configure the temperature preset,
2. Change totally the configuration flow with the help of menu

Normally, all existing temperature will initialize the temperature entities.

<hr /><em>This discussion was created from the release <a href='https://github.com/jmcollin78/versatile_thermostat/releases/tag/6.0.0.alpha1'>Add entities to hold the preset temperature and change configuration menu</a>.</em>

**Comment by jmcollin78** [comment]
See some errors in the log at startup:
```
2024-03-10 19:17:03.843 ERROR (MainThread) [homeassistant.components.number] Platform versatile_thermostat does not generate unique IDs. ID central_configuration_frost_temp is already used by number.central_configuration_frost_temp - ignoring number.central_configuration_frost_temp
```

**Comment by jmcollin78** [comment]
✅  - Temperature entities are well created for central configuration:
![Capture d’écran 2024-03-10 à 19 23 08](https://github.com/jmcollin78/versatile_thermostat/assets/1712374/a310cc66-200f-4237-80a9-4f41158b193f)

✅  - No temperature entities are created when 'use preset from central configuration' is checked:
![Capture d’écran 2024-03-10 à 19 24 05](https://github.com/jmcollin78/versatile_thermostat/assets/1712374/aeae98c1-d594-4183-95e5-e13be1498200)

✅  - New menu is available:
![Capture d’écran 2024-03-10 à 19 25 59](https://github.com/jmcollin78/versatile_thermostat/assets/1712374/1fcc41ad-6817-4aa4-b98f-001ff8de9425)

✅  - When I uncheck 'use central configuration preset', temperature entities are created for the VTherm:
![Capture d’écran 2024-03-10 à 19 26 46](https://github.com/jmcollin78/versatile_thermostat/assets/1712374/1912824b-7e55-4494-942e-14e345eb47a9)
![Capture d’écran 2024-03-10 à 19 28 01](https://github.com/jmcollin78/versatile_thermostat/assets/1712374/f110383e-e532-458c-b733-a665746a9a3f)
✅  -  ... and old values are restored.

✅  - When changing temperature in new temperature entity, the preset change in the VTherm:
![Capture d’écran 2024-03-10 à 19 42 06](https://github.com/jmcollin78/versatile_thermostat/assets/1712374/49fe68cd-ce65-42be-b7ad-4e5886c3a47b)


So, all seems working as expected 🎉

**Reply by jmcollin78** [reply]
Note : i'm not totally sure, but maybe the regulation don't work as expected. I have to check that.

**Reply by jmcollin78** [reply]
No regulation problem constated. I will release the changes soon.

**Comment by Edwintenhaaf** [comment]
Hi @jmcollin78,

I just gave this version a try tonight and I noticed that after a reload of the settings the target temp changes.
It was set at 21 degrees and after the reload if was 20 degrees.
On the stable version this does not happen.

**Reply by jmcollin78** [reply]
Hello thank you for your report. What do you mean "reload of the settings" ? Is that a restart of HA ? Or in "Development Tools"/Yaml reload Versatile Thermostat ?

**Comment by Edwintenhaaf** [comment]
Development Tools/Yaml reload Versatile Thermostat
________________________________
Van: Jean-Marc Collin ***@***.***>
Verzonden: Friday, March 15, 2024 8:10:23 AM
Aan: jmcollin78/versatile_thermostat ***@***.***>
CC: Edwin ten Haaf ***@***.***>; Comment ***@***.***>
Onderwerp: Re: [jmcollin78/versatile_thermostat] Add entities to hold the preset temperature and change configuration menu (Discussion #404)


Hello thank you for your report. What do you mean "reload of the settings" ? Is that a restart of HA ? Or in "Development Tools"/Yaml reload Versatile Thermostat ?

**Reply by jmcollin78** [reply]
Ok. Will see what happens for the release. Have you an idea of where the 20° comes from ?
I understand you change the preset temp to 21° with the new Number entity, then reload all VTherm configuration and then you have 20° in the preset temp of the VTherm (or in the new Number entity) ?

**Comment by Edwintenhaaf** [comment]
I do not use the presets.
Just tried to reproduce the issue and it does not occur when restarting HA only when reloading the integration.
***@***.***

Strange thing however is that after upgrading to the alpha version and the needed restart the issue reproducable every time I reload the settings.
After a second restart the issue disappeared.
I don’t understand why but maybe you do ?

Van: Jean-Marc Collin ***@***.***>
Datum: vrijdag, 15 maart 2024 om 08:37
Aan: jmcollin78/versatile_thermostat ***@***.***>
CC: Edwin ten Haaf ***@***.***>, Comment ***@***.***>
Onderwerp: Re: [jmcollin78/versatile_thermostat] Add entities to hold the preset temperature and change configuration menu (Discussion #404)

Ok. Will see what happens for the release. Have you an idea of where the 20° comes from ?
I understand you change the preset temp to 21° with the new Number entity, then reload all VTherm configuration and then you have 20° in the preset temp of the VTherm (or in the new Number entity) ?

**Reply by Edwintenhaaf** [reply]
still the same issue on stable 6.0.0 I think it's this value in the attributes of the climate device
saved_target_temp: 19.
Target temp is 20 but on reload it reverts back to this value

**Reply by jmcollin78** [reply]
I try to reproduce your case without success.
Can you please describe the reproduction steps, step by step. I have to admit the process leading to the issue is not clear to me.

**Comment by Edwintenhaaf** [comment]
Sorry, I will try to explain.

Using expert mode fine tuning the pid settings.
(Difficult because the ac only accepts steps of 1c)

When I change the settings in the yaml config I go to developers tools > services > Versatile Thermostat configuration: Reload
After the reload the target temperature drops from 20 to 19

Does this makes sense ?




Van: Jean-Marc Collin ***@***.***>
Datum: dinsdag, 19 maart 2024 om 09:59
Aan: jmcollin78/versatile_thermostat ***@***.***>
CC: Edwin ten Haaf ***@***.***>, Comment ***@***.***>
Onderwerp: Re: [jmcollin78/versatile_thermostat] Add entities to hold the preset temperature and change configuration menu (Discussion #404)

I try to reproduce your case without success.
Can you please describe the reproduction steps, step by step. I have to admit the process leading to the issue is not clear to me.

**Reply by jmcollin78** [reply]
It is more clear and I will try to reproduce. Most people, uses preset and I'm sure it is working as expected with a preset selected.

Normally, in manual mode (because you don't use the preset right ?), after restart, the target temperature is restored from its last state. But with V6.0 all of this code have been deeply updated, so I will have a closer look to this part.

Why don't you use presets ? Is there a good reason ?
